stats/stats.go

```diff
@@ -55,52 +55,61 @@ func ConditionallyStartStatsServer(addhandlers ...func(mux *http.ServeMux, profi
 }
 
 func StartStatsServerWithPort(startupOpts StartupOptions, addhandlers ...func(mux *http.ServeMux, profiles map[string]string)) {
+	StartCancellableStatsServerWithPort(context.Background(), startupOpts, addhandlers...)
+}
+
+func StartCancellableStatsServerWithPort(ctx context.Context, startupOpts StartupOptions, addhandlers ...func(mux *http.ServeMux, profiles map[string]string)) {
 	// if the env var was provided (i.e., startup is conditional) and the value of that env var is not the expected value, then return and do nothing
 	if startupOpts.EnvVar != "" && os.Getenv(startupOpts.EnvVar) != startupOpts.EnabledValue {
 		return
 	}
 
-	var logLevel zap.AtomicLevel
-	if envLogLevel := os.Getenv("LOG_LEVEL"); envLogLevel != "" {
-		var setLevel zapcore.Level
-		switch envLogLevel {
-		case "debug":
-			setLevel = zapcore.DebugLevel
-		case "warn":
-			setLevel = zapcore.WarnLevel
-		case "error":
-			setLevel = zapcore.ErrorLevel
-		case "panic":
-			setLevel = zapcore.PanicLevel
-		case "fatal":
-			setLevel = zapcore.FatalLevel
-		default:
-			setLevel = zapcore.InfoLevel
-		}
+	if envLogLevel := os.Getenv(contextutils.LogLevelEnvName); envLogLevel != "" {
+		contextutils.SetLogLevelFromEnv(envLogLevel)
+	}
 
-		contextutils.SetLogLevel(setLevel)
+	mux := new(http.ServeMux)
```

Comment:
You touched on it a little in slack, but could you clarify why we're moving this mux setup out of the goroutine?

Reply:
I wanted to be able to have another goroutine that could shutdown the server. I ran into the issue when writing tests that a single test would pass, but running multiple would fail, because the original goroutine was leaking / still running.

i'm going to move this construction back to the goroutine, back to how it used to be